File: Escola/Cursos.Api/scripts/service/temaService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/13/2017 1:58:38 PM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("TemaService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Tema/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Tema/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Tema, callback) {
            $http.post(serviceBase + "Tema/post", Tema)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Tema, callback) {
            $http.post(serviceBase + "Tema/put", Tema)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Tema, callback) {
            $http.post(serviceBase + "Tema/delete", Tema)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

File: Escola/Cursos.Repository/Services/TemaService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  TemaService
      /// </summary>
      public partial class TemaService : BaseServices, ITemaService
      {

		public TemaService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity EV_TEMA
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<EV_TEMA> entity)
        {
            UnitOfWork.TemaRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity EV_TEMA
        /// </summary>
        /// <param name="entity"></param>
		public void Add(EV_TEMA entity)
        {	
            UnitOfWork.TemaRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity EV_TEMA
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(EV_TEMA entity)
        {
            UnitOfWork.TemaRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity EV_TEMA
        /// </summary>
        /// <returns>IEnumerable EV_TEMA</returns>
        public IEnumerable<EV_TEMA> GetAll()
        {
			return UnitOfWork.TemaRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity EV_TEMA
        /// </summary>
        /// <param name="entity"></param>
        public void Update(EV_TEMA entity)
        {
            UnitOfWork.TemaRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Get Entity EV_TEMA By Id
        /// </summary>
        /// <param name="id"></param>
        /// <returns>EV_TEMA</returns>
		public EV_TEMA GetById(int id)
        {
			return UnitOfWork.TemaRepository.FindBy(x => x.ID.Equals(id)).SingleOrDefault();
        }		
				
		public IEnumerable<EV_TEMA> FindBy(Expression<Func<EV_TEMA, bool>> predicate)
        {
            return UnitOfWork.TemaRepository.FindBy(predicate);
        }
      }
}
